app/Http/Controllers/Api/SignupsController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Rogue\Http\Controllers\Api;
+
+use Illuminate\Http\Request;
+use Rogue\Services\SignupService;
+use Rogue\Repositories\PhotoRepository;
+use Rogue\Http\Transformers\SignupTransformer;
+
+class SignupsController extends ApiController
+{
+    /**
+     * @var \League\Fractal\TransformerAbstract;
+     */
+    protected $transformer;
+
+    /**
+     * The signup service instance.
+     *
+     * @var Rogue\Services\SignupService
+     */
+    protected $signups;
+
+    /**
+     * The photo repository instance.
+     *
+     * @var Rogue\Repositories\PhotoRepository
+     */
+    protected $photos;
+
+    /**
+     * Create a controller instance.
+     *
+     * @param  PostContract  $posts
+     * @return void
+     */
+    public function __construct(SignupService $signups, PhotoRepository $photos)
```

Comment:
We should be using the new `PostService` instead of the `PhotoRepostitory` now. The `PostService` would resolve which repository to use to create the resource based on the `event_type` passed in the request.